lib/p2p/Peer.ts

```diff
@@ -464,14 +465,46 @@ class Peer extends EventEmitter {
       const startTime = Date.now();
       let retryDelay = Peer.CONNECTION_RETRIES_MIN_DELAY;
       let retries = 0;
-
-      this.socket = net.connect(this.address.port, this.address.host);
       this.inbound = false;
       this.connectionRetriesRevoked = false;
 
+      const connectViaProxy = () => {
+        const proxyOptions = {
```

Comment:
I think it's a bit better to do `const proxyOptions: SocksClientOptions = {` here to check that our types are correct rather than `as SocksClientOptions` further below. It'll require to add the `!` operator to `torSocks` or to make an upfront check to make sure `torSocks` has a value, but I think it'd be more robust going forward that way.